src/cascadia/TerminalControl/TermControlAutomationPeer.cpp

```diff
@@ -112,7 +112,7 @@ namespace winrt::Microsoft::Terminal::Control::implementation
         {
             if (const auto keyEventChar{ gsl::narrow_cast<wchar_t>(charCode) }; IsReadable({ &keyEventChar, 1 }))
             {
-                _keyEvents.emplace_back(keyEventChar);
+                _keyEvents.lock()->emplace_back(keyEventChar);

```

Comment:
@lhecker this is an acceptable use of lock(), right?

Reply:
Yeah the guard should get destroyed at `;` (= after the insertion finished).